baseapp/abci.go

```diff
@@ -179,6 +179,9 @@ func (app *BaseApp) MidBlock(ctx sdk.Context, height int64) (events []abci.Event
 
 // EndBlock implements the ABCI interface.
 func (app *BaseApp) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) (res abci.ResponseEndBlock) {
+	// Clear DeliverTx Events
+	ctx.MultiStore().ResetEvents()
```

Comment:
Clear it in endBlock when we're done with the deliverTx events

Reply:
Maybe we don't really need to clear the events here and just let it get garbage collected once the block is committed since it's not interblock store but mainly reset for checktx (runs synchronously) so we dont OOM